src/vm/appdomain.cpp

```diff
@@ -12956,6 +12956,11 @@ void AppDomain::CreateADUnloadWorker()
         DWORD stackSize = 512 * 1024; // leave X64 unchanged since we have plenty of VM
 #endif
         Thread *pThread = SetupUnstartedThread();
+        if (pThread == NULL)
+        {
+            ThrowOutOfMemory();            
```

Comment:
I was originally going to do that, but ended up preferring the other way. Only two places needed to add the throw, there was one that got simplified (removing the catch), and two where throwing the exception would not be expected.